### PR TITLE
Use Edit aka Levenshtein distance in string comparator

### PR DESCRIPTION
**Pull request by albertaleksieiev**
### Motivation
Previously comparing do not handle typo, i.e. **J`e`nsen Intercept`o`r** and  **J`o`nsen Intercept`a`r** was be not an equal at all. 

### Changes
Changed only word matching algorithm - [edit distance](https://en.wikipedia.org/wiki/Levenshtein_distance), and tests is added. 

### Result
Now comparing is handle typo. If some of the letters do not match, the result will be lower(0 - words is totally different, 100 - words is equal).
 **J`e`nsen Intercept`o`r** and  **J`o`nsen Intercept`a`r**,  need to change 2 letters, so distance will be 2. But the result will be
```
MAX DISTANCE = is sum of letters in first word(18)
(1 - EDIT_DISTANCE / MAX_DISTANCE ) * 100 = (1 - 2 / 18) * 100 = 88.89
```

### CODE REVIEW

File: src/test/java/songbox/house/util/StringByEachWordComparatorTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.junit.Test;
+import songbox.house.util.compare.StringByEachWordComparator;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static songbox.house.util.compare.StringByEachWordComparator.MAX_COMPARE_RESULT;
+
+public class StringByEachWordComparatorTest {
```
Comment (Denstden) [inline src/test/java/songbox/house/util/StringByEachWordComparatorTest.java]: Very good

File: src/main/java/songbox/house/util/compare/StringByEachWordComparator.java [hunk 2]
```diff
@@ -4,25 +4,77 @@
 import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class StringByEachWordComparator implements Comparator<String> {
```
Comment (Denstden) [inline src/main/java/songbox/house/util/compare/StringByEachWordComparator.java]: Maybe rename class to LevenshteinDistanceComparator?

File: src/test/java/songbox/house/util/StringByEachWordComparatorTest.java (hunk 1 above)
Comment (albertaleksieiev) [reply src/test/java/songbox/house/util/StringByEachWordComparatorTest.java]: @Denstden sure!

File: src/test/java/songbox/house/util/StringByEachWordComparatorTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                .sum();
+
+        assertEquals((int) ((1. - (float) expectedDistance / maxDistance) * MAX_COMPARE_RESULT), comparator.compare(words1, words2));
+    }
+
+    @Test
+    public void shouldCompareWord() {
+        // Remove `l` from first word(Ivan), and change `o` into `a` on second word(Brody)
+        // Expected distance 2
+        testComparing("Ivan Brody", "Ival Brodsady Brady", 2);
```
Comment (Denstden) [inline src/test/java/songbox/house/util/StringByEachWordComparatorTest.java]: JUnit has @Parameterized feature

https://www.tutorialspoint.com/junit/junit_parameterized_test.htm

Comment (albertaleksieiev) [reply src/test/java/songbox/house/util/StringByEachWordComparatorTest.java]: @Denstden woah 💂🏼This feature looks interesting, I'll try it 👌

**Pull request merged**